tools/RepositoryManagement/DevHome.RepositoryManagement/Services/EnhanceRepositoryService.cs

```diff
@@ -0,0 +1,170 @@
+            return repository.Head.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting branch name");
+        }
+
+        return string.Empty;
+    }
+
+    public string GetRepositoryUrl(string repositoryLocation)
+    {
+        try
+        {
+            var repository = new LibGit2Sharp.Repository(repositoryLocation);
+            return repository.Network.Remotes.First().Url;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting the repositoryUrl");
+        }
+
+        return string.Empty;
+    }
+
+    public IPropertySet GetProperties(string[] propertiesToReturn, string repositoryLocation)
+    {
+        var sourceControlProvider = new SourceControlProvider();
+        var provider = sourceControlProvider.GetProvider(repositoryLocation);
+
+        if (provider == null)
+        {
+            _log.Warning($"Path: {repositoryLocation} does not have an associated provider.");
+            return new PropertySet();
+        }
+
+        // This call does check the settings for file explorer and souce control integration.
+        return provider.GetProperties(propertiesToReturn, string.Empty);
+    }
+
+    private async Task<string> GetRepositoryLocationFromUser()
+    {
+        StorageFolder repositoryRootFolder = null;
+        try
+        {
+            using var folderDialog = new WindowOpenFolderDialog();
+            repositoryRootFolder = await folderDialog.ShowAsync(_window);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error occurred when selecting a folder for adding a repository.");
+            _telemetry.LogError(
+                ErrorEventName,
+                LogLevel.Critical,
+                new EnhanceRepositoryErrorEvent(nameof(SelectRepositoryAndMakeItEnhanced), ex.HResult, ex.Message, string.Empty));
+            return string.Empty;
+        }
+
+        if (repositoryRootFolder == null || repositoryRootFolder.Path.Length == 0)
+        {
+            _log.Information("Didn't select a location to register");
+            return string.Empty;
+        }
+
+        _log.Information($"Selected '{repositoryRootFolder.Path}' as location to register");
+        return repositoryRootFolder.Path;
+    }
+
+    private async Task<Guid> AssignSourceControlToPath(string maybeRepositoryPath)
+    {
+        if (string.IsNullOrEmpty(maybeRepositoryPath))
+        {
+            _log.Information("maybeRepositoryPath is null or empty.  Not assigning a source control provider");
+            return Guid.Empty;
+        }
+
+        if (!Directory.Exists(maybeRepositoryPath))
+        {
+            Directory.CreateDirectory(maybeRepositoryPath);
+        }
+
+        foreach (var extension in _repositorySourceControlProviders)
+        {
+            var didAdd = await _sourceControlRegistrar.AssignSourceControlProviderToRepository(extension, maybeRepositoryPath);
```

Comment:
"didAdd" feels a little weird to me because it implies success (it did the add) but when we check success on the next line. Maybe use something more neutral?

Reply:
Changed to `assignSourceControlResult`